src/rust/src/backend/ec.rs

```diff
@@ -274,14 +274,35 @@ impl ECPrivateKey {
             ));
         }
 
-        let (data, _) = utils::calculate_digest_and_algorithm(
+        let (data, _algo) = utils::calculate_digest_and_algorithm(
```

Comment:
if this is used, the name shouldn't have a leading prefix

Reply:
it's only used when OpenSSL >=3.2.0, so without it it complains when compiling with older OpenSSLs (or libre/boring). I could do something like this instead?:
```rust
cfg_if::cfg_if! {
    if #[cfg(CRYPTOGRAPHY_OPENSSL_320_OR_GREATER)]{
        let (data, _algo) = utils::calculate_digest_and_algorithm(
            py,
            data.as_bytes(),
            signature_algorithm.getattr(pyo3::intern!(py, "algorithm"))?,
        )?;
    } else {
        let (data, _) = utils::calculate_digest_and_algorithm(
            py,
            data.as_bytes(),
            signature_algorithm.getattr(pyo3::intern!(py, "algorithm"))?,
        )?;
    }    
````

Reply:
You can do `let _ = algo;` in the other branch.

Reply:
oh, that is a lot better! fixed